networkx/algorithms/community/label_propagation.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 02 14:03:50 2014
+Based on the code donated by Tyler Rush 
+(https://github.com/networkx/networkx/issues/617)
+
+LPA community detection algorithm (based on 'Community Detection via 
+Semi-Synchronous Label Propagation Algorithms' Cordasco and Gargano, 2011
+
+@author: Aitor Almeida <aitoralmeida@gmail.com>
+All rights reserved.
+BSD license.
```

Comment:
Copyright information does not belong in documentation strings, it belongs in a comment at the head of the file.